meson.build

```
@@ -0,0 +1,89 @@
+project(
+  'pymsis',
+  'c',
+  # Note that the git commit hash cannot be added dynamically here
+  version: '0.6.0',
+  license: 'MIT',
+  meson_version: '>= 0.63.3',
+  default_options: [
+    'buildtype=debugoptimized',
+    'c_std=c99',
+    'fortran_std=legacy',
+  ],
+)
+
+cc = meson.get_compiler('c')
+
+# TODO: the below -Wno flags are all needed to silence warnings in
+# f2py-generated code. This should be fixed in f2py itself.
+_global_c_args = cc.get_supported_arguments(
+  '-Wno-unused-but-set-variable',
+  '-Wno-unused-function',
+  '-Wno-conversion',
+  '-Wno-misleading-indentation',
+  '-Wno-incompatible-pointer-types',
+)
+add_project_arguments(_global_c_args, language : 'c')
+
+
+# Adding at project level causes many spurious -lgfortran flags.
+add_languages('fortran', native: false)
+ff = meson.get_compiler('fortran')
+if ff.has_argument('-Wno-conversion')
+  add_project_arguments('-Wno-conversion', language: 'fortran')
+endif
+
+is_windows = host_machine.system() == 'windows'
+# Platform detection
+is_mingw = is_windows and cc.get_id() == 'gcc'
+if is_windows
+  # Need static libraries to get libgfortran
+  gcc_link_args = ['-static']
+  if is_mingw
+    add_project_link_arguments(gcc_link_args, language: ['c'])
+  endif
+  if meson.get_compiler('fortran').get_id() == 'gcc'
+    add_project_link_arguments(gcc_link_args, language: ['fortran'])
+  endif
+endif
+
+generate_f2pymod = files('tools/generate_f2pymod.py')
+
+# https://mesonbuild.com/Python-module.html
+py_mod = import('python')
+py3 = py_mod.find_installation()
```

```suggestion
py3 = py_mod.find_installation(pure: false)
```

https://mesonbuild.com/Release-notes-for-0-64-0.html#pythonfind_installation-now-accepts-pure-argument